File: GCD of two Numbers.js
// class Solution {
//     gcd(a, b) {
//         // code here
//         let c1=[]
//         for (let i=1;i<=Math.min(a,b);i++){
//             if (a%i===0 && b%i===0){
//                 c1.push(i)
//             }
//         }
//         return Math.max(...c1)
//     }
// }
// console.log(new Solution().gcd(4,6))

// another
class Solution{
    gcd(a,b){
result=1
        while (b!==0){
            let temp=b
            b=a%b
            a=temp
        }
        return a
    }
}
console.log(new Solution().gcd(4,6))

File: GCD of two Numbers.py
class Solution:
    def GCD(self, n1, n2):
        count1=[]
        count2=[]
        for i in range(1,n1+1):
            if n1%i==0:
                count1.append(i)
        for j in range(1,n2+1):
            if n2%j==0:
                count2.append(j)
        return max(set(count1) & set(count2))
print(Solution().GCD(4,6))
